### ISSUE TITLE
expiration date on student profiles?

### ISSUE THREAD

**Issue by meagansiu** [maltyeva/mindimp-pg#91]

**Comment by maltyeva** [comment]
Good idea -- So the easiest thing would be to have students either be 'active' or 'inactive' on the site, which I could probably do today or tomorrow. 

Another option would be to add an expiration date and then every time a student logs in, which whether the expiration date is after today's date and if it is, lock them out. Would this implementation be for the student or for the teachers though? Would we want to send them warnings when their profile is about to expire? What should the actions to instructors look like when a student is de-activated or their account expires? 

Thoughts? :)

**Comment by ChasonMindImp** [comment]
I think this may be as much (or more) of an admissions/enrollment management question than a technical one. Trying to think about it from a student experience and administrative perspective, activity and inactivity will vary depending on enrollment status and other factors. Each one may be worth addressing, in part because some of those status will be making us money and drumming up good student engagement, and others won't. From my seat, this is what this could all look like, and various things we should consider when it comes to levels of access. This is definitely not exhaustive, but these are some things that come to mind. 

Active/No Program
- Status: Student has been accepted to MindImp, receives orientation materials and directives, is able to set up an online account and does so to satisfactory completion. Student has not been assigned to course or reading program.
- When this happens: Immediately upon acceptance
- Access: Profile creation
- Notes: Resolving this status is easily fixed if we require students to elect one or more programs at the time of acceptance. It will be easier from an administrative perspective because it is one less student experience to work around and deal with, and it puts students one step closer to enrolling in a program and paying us money. 

Active/No Enrollment
- Status: Student has been accepted to MindImp, receives orientation materials and directives, is able to set up an online account and does so to satisfactory completion. Student has been assigned to the reading and/or course program. Student has not been assigned an advisor and reading/is not signed up for a course. 
- When this happens: Between program election and registration, down-time between one program or course ending and beginning another. 
- Access: Profile creation, view article list, view book list, view course list, add to wishlist, see advisor profiles, completed work and responses (if they've previously completed course/reading work)
- Notes: For this status, we want to tantalize students with the possibilities. They should be able to get a sense of what they can do, without actually being able to do it. From an enrollment management perspective, this will need to be coupled with engagement to encourage students to enroll again, with little downtime in between. 

Active/Enrolled
- Status: Student has been accepted to MindImp, receives orientation materials and directives, is able to set up an online account and does so to satisfactory completion. Student has been assigned to the reading and/or course program. Student has been assigned an advisor and readiing/is signed up for a course. 
- Access:Profile creation, view lists, add to wishlist, have done/read lists, advisor profiles, completed work and responses, full access to materials necessary to complete assignments for enrolled program
- Notes: This will be the most straightforward and most accessible. Also where we want as many of our students to be as possible at any given time. 

Inactive
- Status: Student actively requests a period of inactivity (a leave of absence, if you will) for a specified or indefinite amount of time. Student may also be reverted to inactive status if they remain Active/No Enrollment for a period of time determined by MindImp.
- Access: Profile creation, program status, previous work. Unsure if we want list view as with Active/No Enrollment
- Notes: Inactivity tends to be difficult. Technically, depending on what we want, it could look almost indistinguishable from Active/No Enrollment, and we could set it up to be such. The big difference there may just be courting students back to the program,depending on if they elect inactivity or fall into it by default. 
- More notes: Students do not need to reapply to return to active status. However, if inactivity is excessive or unplanned (as when a student will call us and say they need to take XX time off to attend a camp, etc.), then requiring something, such as a short essay stating intent to return of a short English re-evaluation, could be useful and/or necessary.

Program Complete
- Status: Student leaves MindImp with no intention of participating in future offerings
- When this happens: At student discretion or after a period of inactivity to be determined by MindImp (i.e. if student remains inactive or X year(s), then they are switched to Program Complete)
- Access: None
- Notes: At this point, students become alumni. Access to profile, community, previous work, and the like could be very useful for us for engagement, marketing, feedback, and the like. However, I think that is an entirely separate conversation. 
- More notes: Students who wish to return to MindImp must reapply. 

We want to toe the line of making sure access is appropriate with what students are doing without unnecessarily closing the door to students who think they may want to return. Coupling this with enrollment management (as an extension of admissions policy) will also keep us in consideration with students who are active/unenrolled or inactive.

To turn it back to a technical discussion, I think we need to, if possible, finalize internal terms and policies for these designations and then have each as an option on the site. Admissions staff will then have the task of toggling students as they fall in and out of certain statuses.

**Comment by maltyeva** [comment]
Good comments!

Okay, so for now, I definitely want an active or an inactive 'toggle', because when doing things like choosing which students are in your classes, you want to only be able to select students who are active on the site (and maybe those who are in the course permissions list?), but I do agree that well need to have stronger definitions of what it means to be a student on the website. What would be the best way to go forth with establishing that procedure? 

One potential downside of having students choose their program right away is if we transition from having the wenjuan page to having students do their inquiry on the site instead, because they may not be familiar with the programs (we could have a choice for preferred in the beginning at first?), because in the current implementation, they are given access to the program with the toggle. 

I do like the idea of calling deactivated students Alumni. At this point, will we have students who are completely finished with the program after this class session?

**Comment by jzthompson** [comment]
We have plenty of students who will not be immediately continuing with courses in the next academic term, but I don't think we have anyone who is unequivocally and permanently done with all MindImp programs. In other words, we have a number of students who at least for a while will fit into the "Inactive" category. I like the idea of trying to tempt these students back into the program.  I wonder if periodic email updates (Added five new books, ten new articles this month, starting three new courses next month, etc.) would also be a good way to keep them involved with the website.

**Comment by meagansiu** [comment]
I think that the periodic email update sounds cool! I get some emails from
Coursera about their new offerings that look pretty nice; perhaps we could
ask one of the graphic designers to create something we could use, so that
it looks more MindImp-branded instead of just a plain text email.

On Thu, Apr 23, 2015 at 9:28 PM, jzthompson <notifications@github.com>
wrote:

> We have plenty of students who will not be immediately continuing with
> courses in the next academic term, but I don't think we have anyone who is
> unequivocally and permanently done with all MindImp programs. In other
> words, we have a number of students who at least for a while will fit into
> the "Inactive" category. I like the idea of trying to tempt these students
> back into the program. I wonder if periodic email updates (Added five new
> books, ten new articles this month, starting three new courses next month,
> etc.) would also be a good way to keep them involved with the website.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/maltyeva/mindimp-pg/issues/91#issuecomment-95584595>.
>



-- 
Meagan Siu, B.A. | 萧笑嫣
Program Manager and Advisor* | **思脉**MindImp *
http://www.mindimp.com.cn/
https://www.mindimp.com/
